strider/query_planner.py

```diff
@@ -195,6 +188,23 @@ async def find_valid_permutations(
 
     expanded_qg = expand_qg(qgraph, logger)
 
+    reverse_edges = {}
+    # Add reverse edges so that we can traverse edges in both directions
+    for edge_id, edge in expanded_qg['edges'].items():
+        reverse_edge = edge.copy()
+        reverse_edge['predicate'] = [
+            f"<-{p}-" for p in reverse_edge['predicate']]
+        # Swap subject and object
+        reverse_edge['subject'], reverse_edge['object'] = \
+            reverse_edge['object'], reverse_edge['subject']
+        reverse_edges[f"{edge_id}{REVERSE_EDGE_SUFFIX}"] = reverse_edge
```

Comment:
Admirable loyalty to f-strings.
```suggestion
        reverse_edges[edge_id + REVERSE_EDGE_SUFFIX] = reverse_edge
```